app/services/hyrax/solr_query_service.rb

```diff
@@ -0,0 +1,162 @@
+      self
+    end
+
+    ##
+    # @param model [Class|String] a class from the model (e.g. Hyrax::Work, Hyrax::FileSet, etc.)
+    # @return [SolrQueryService] the existing service with model query appended
+    def with_model(model:)
+      raise ArgumentError, "Expected <Class> or <String> for model:, but received <#{model.class}>" unless model.is_a?(Class) || model.respond_to?(String)
+      model_query = construct_query_for_model(model)
+      @query += [model_query]
+      self
+    end
+
+    ##
+    # @param generic_type [String] (Default: Work)
+    # @return [SolrQueryService] the existing service with model query appended
+    def with_generic_type(generic_type: 'Work')
+      # TODO: Generic type was originally stored as `sim`.  Since it is never multi-valued, it is moving to being stored
+      #       as `si`.  Until a migration is created to correct existing solr docs, this query searches in both fields.
+      field_pairs = { generic_type_si: generic_type, generic_type_sim: generic_type }
+      type_query = construct_query_for_pairs(field_pairs, ' OR ', 'field')
+      @query += [type_query]
+      self
+    end
+
+    ##
+    # @param field_pairs [Hash] a list of pairs of property name and values (e.g. { field1: values, field2: values })
+    # @param join_with [String] the connector used to join the field pairs (default: ' AND ')
+    # @param type [String] type of query to run. Either 'raw' or 'field' (default: 'field')
+    # @return [SolrQueryService] the existing service with field_pair query appended
+    def with_field_pairs(field_pairs: {}, join_with: default_join_with, type: 'field')
+      pairs_query = construct_query_for_pairs(field_pairs, join_with, type)
+      return self if pairs_query.blank?
+      @query += [pairs_query]
+      self
+    end
+
+    private
+
+    # Construct a solr query for a list of ids
+    # @param [Array] ids to be included in the query
+    # @return [String] a solr query
+    # @example
+    #   construct_query_for_ids(['id1', 'id2'])
+    #   # => "{!terms f=id}id1,id2"
+    def construct_query_for_ids(ids)
+      "{!terms f=#{Hyrax.config.id_field}}#{ids.join(',')}"
+    end
+
+    # Construct a solr query from a list of pairs (e.g. { field1: values, field2: values })
+    # @param [Hash] field_pairs a list of pairs of property name and values
+    # @param [String] join_with the value we're joining the clauses with (default: ' AND ')
+    # @param [String] type of query to run. Either 'raw' or 'field' (default: 'field')
+    # @return [String] a solr query
+    # @example
+    #   construct_query([['library_id_ssim', '123'], ['owner_ssim', 'Fred']])
+    #   # => "_query_:\"{!field f=library_id_ssim}123\" AND _query_:\"{!field f=owner_ssim}Fred\""
+    def construct_query_for_pairs(field_pairs, join_with = default_join_with, type = 'field')
+      clauses = pairs_to_clauses(field_pairs, type)
+      return "" if clauses.count.zero?
+      return clauses.first if clauses.count == 1
```

Comment:
is this line to keep the parentheses wrap out? i think the internal string is the same, either way, so i was wondering if this conditional is necessary.

Reply:
I copied this over from Hyrax::SolrQueryBuilderService without examining it to determine if it should be changed.  I'm not sure of the thoughts of the original author.

Reply:
I looked at the original ActiveFedora version of `construct_query` and it is quite different.  So that isn't a help.

But looking at the code, I agree with you assessment that the purpose of line 109 is to return the clause without parentheses if there is only one field pair.